src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -667,189 +598,38 @@ public enum ConnectionType
-                using (var writer = new BlittableJsonTextWriter(documentsContext, tcpConnectionOptions.Stream))
-                using (documentsContext.OpenReadTransaction())
-                using (configurationContext.OpenReadTransaction())
-                {
-                    string changeVector = null;
-                    long lastEtagFromSrc = 0;
-
-                    if (getLatestEtagMessage.ReplicationsType != ReplicationLatestEtagRequest.ReplicationType.Migration &&
-                        ShardHelper.IsShardedName(Database.Name) == false)
-                    {
-                        changeVector = DocumentsStorage.GetFullDatabaseChangeVector(documentsContext);
-
-                        lastEtagFromSrc = DocumentsStorage.GetLastReplicatedEtagFrom(
-                        documentsContext, getLatestEtagMessage.SourceDatabaseId);
-                    if (_log.IsInfoEnabled)
-                        _log.Info($"GetLastEtag response, last etag: {lastEtagFromSrc}");
-                    }
-
-                    var response = new DynamicJsonValue
-                    {
-                        [nameof(ReplicationMessageReply.Type)] = nameof(ReplicationMessageReply.ReplyType.Ok),
-                        [nameof(ReplicationMessageReply.MessageType)] = ReplicationMessageType.Heartbeat,
-                        [nameof(ReplicationMessageReply.LastEtagAccepted)] = lastEtagFromSrc,
-                        [nameof(ReplicationMessageReply.NodeTag)] = _server.NodeTag,
-                        [nameof(ReplicationMessageReply.DatabaseChangeVector)] = changeVector,
-                        [nameof(ReplicationMessageReply.AcceptablePaths)] = replParams?.AllowedPaths,
-                        [nameof(ReplicationMessageReply.PreventDeletionsMode)] = replParams?.PreventDeletionsMode
-                    };
-
-                    documentsContext.Write(writer, response);
-                    writer.Flush();
-                }
-            }
-            catch (Exception)
-            {
-                try
-                {
-                    tcpConnectionOptions.Dispose();
-                }
-                catch (Exception)
-                {
-                    // do nothing
-                }
-
-                throw;
-            }
-
-            if (_log.IsInfoEnabled)
-                _log.Info(
-                    $"Initialized document replication connection from {connectionInfo.SourceDatabaseName} located at {connectionInfo.SourceUrl}");
-
-            return getLatestEtagMessage;
-        }
-
-        private long _reconnectInProgress;
+        internal static readonly TimeSpan MaxInactiveTime = TimeSpan.FromSeconds(60);
 
-        private void ForceTryReconnectAll()
+        protected override DynamicJsonValue GetInitialRequestMessage(ReplicationLatestEtagRequest getLatestEtagMessage, PullReplicationParams replParams = null)
```

Comment:
the name should be `GetResponseToReplicationLatestEtagRequest` or something like that? you are not getting here the initial request message, right?